lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py

```diff
@@ -0,0 +1,285 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: cloudfront_origin_access_identity
+short_description: create, update and delete origin access identities for a
+                   cloudfront distribution.
+description:
+    - Allows for easy creation, updating and deletion of origin access
+      identities.
+requirements:
+  - boto3 >= 1.0.0
+  - python >= 2.6
+version_added: "2.4"
+author: Willem van Ketwich (@wilvk)
+options:
+    state:
+      description:
+        - The state of the resource. Valid states are
+            present
+            absent
+      required: false
+      default: present
+    origin_access_identity_id:
+      description:
+        - The origin_access_identity_id of the cloudfront distribution.
+      required: false
+    comment:
+      description:
+        - A comment to describe the cloudfront origin access identity.
+      required: false
+    caller_reference:
+      description:
+        - A unique identifier to reference the origin access identity by.
+      required: false
+
+'''
+
+EXAMPLES = '''
+
+# create an origin access identity
+
+- cloudfront_origin_access_identity:
+    state: present
+    caller_reference: this is an example reference
+    comment: this is an example comment
+
+# update an existing origin access identity
+# using caller_reference as an identifier
+
+- cloudfront_origin_access_identity:
+     state: present
+     origin_access_identity_id: E17DRN9XUOAHZX
+     caller_reference: this is an example reference
+     comment: this is a new comment
+
+# delete an existing origin access identity
+# using caller_reference as an identifier
+
+- cloudfront_origin_access_identity:
+     state: absent
+     caller_reference: this is an example reference
+     comment: this is a new comment
+
+'''
+
+RETURN = '''
+
+location:
+    description: describes a url specifying the origin access identity.
+    returned: always
+    type: str
+
+'''
+
+from ansible.module_utils.ec2 import get_aws_connection_info, ec2_argument_spec
+from ansible.module_utils.ec2 import boto3_conn, HAS_BOTO3
+from ansible.modules.cloud.amazon.cloudfront_facts import (
+    CloudFrontFactsServiceManager)
+import ansible.module_utils.cloudfront as helpers
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+from ansible.module_utils.basic import AnsibleModule
```

Comment:
It would be better if AnsibleAWSModule was used instead of AnsibleModule because the exception handling is better (and cloudfront_distribution is using it). To import: `from ansible.module_utils.aws.core import AnsibleAWSModule`

Instead of using e.response and str(e) yourself, you can call 
```
except (ClientError, BotoCoreError) as e:
    module.fail_json_aws(e, msg="Unable to <thing that failed>")
```
and it formats the exception by using e.message if it exists or str(e) and uses e.response only if it exists.

Reply:
Using AnsibleAWSModule also means you can remove importing HAS_BOTO3 and the HAS_BOTO3 check, as the module checks it.